contributing/publishing.md

```diff
@@ -7,42 +7,77 @@ extensions through the .vsix files.
+To create a release branch manually:
 
-```json
-{
-  "event_type": "create_release_branch",
-  "client_payload": {
-    "version": "xx.yy.zz"
-  }
-}
-```
+<b>Note that this isn't typically required due to the scheduled job in CircleCi</b>
 
-You can test this using CURL like so:
+1. Open the Command Palette (press Ctrl+Shift+P on Windows or Linux, or Cmd+Shift+P on macOS).
+1. Search for `Tasks: Run Task`.
+1. Select `Create Release Branch`.
 
-```bash
-curl -H 'Authorization: Bearer [your GitHub personal access token]' \
--X POST -d "{\"event_type\": \"create_release_branch\", \"client_payload\": {\"version\": \"48.4.1\"}}" \
-https://api.github.com/repos/forcedotcom/salesforcedx-vscode/dispatches
-```
+## Generating the Change Log
+
+We generate the changelog based off of the new commits that are being staged for production. The change log generator helps us automate the process of generating the `CHANGELOG.md` with the correct format and commits being staged.
+
+To run the change log generator:
+
+1. Run `git pull` to make sure your local changes are up to date.
+1. Open the Command Palette (press Ctrl+Shift+P on Windows or Linux, or Cmd+Shift+P on macOS).
+1. Search for `Tasks: Run Task`.
+1. Select `Create Change Log`.
+
+## Merging the Release Branch into Main
+
+After the change log has been approved and merged into your release branch, it's time to prepare main with the new changes for the publish. We currently utilize a CircleCi workflow that will rebase main off of the release branch. We are specifically using the rebase strategy because we want all of the commits from our release branch to be applied ontop of the commits in main.
+
+To run the merge process:
+
+1. Open the Command Palette (press Ctrl+Shift+P on Windows or Linux, or Cmd+Shift+P on macOS).
+1. Search for `Tasks: Run Task`.
+1. Select `Launch Pre-Publish Steps`.
+1. Approve the workflow in CircleCi:
+   1. Navigate to the `#pdt_releases` channel in Slack.
+   1. There should soon be a `Pending Approval for Publish` option. Click the `Visit Workflow` button to navigate to CircleCi.
+   1. Click the selection for `hold`.
+   1. You should now see the following ![Approval View](./images/contributing-approval-button.png). Click the `Approve` button.
+
+## Publishing Main
+
+After the pre-publish steps have run and main has been rebased off of the release branch, it's now time to publish main.
+
+1. Open the Command Palette (press Ctrl+Shift+P on Windows or Linux, or Cmd+Shift+P on macOS).
+1. Search for `Tasks: Run Task`.
+1. Select `Publish Extensions`.
+1. Approve the workflow in CircleCi:
+   1. Navigate to the `#pdt_releases` channel in Slack.
+   1. There should soon be a `Pending Approval for Publish` option. Click the `Visit Workflow` button to navigate to CircleCi.
+   1. Click the selection for `hold`.
+   1. You should now see the following ![Approval View](./images/contributing-approval-button.png). Click the `Approve` button.
+
+# Manual Publish
```

Comment:
I'd like to get Luis' feedback on this piece. It has been a long time since we've had to do the process manually, so I'm not sure how much of this applies anymore.

Reply:
It's a good option to keep it available in the situation ex. CircleCI is down

Reply:
The steps are still valid but we need to remove the old AWS steps, here's a PR for it https://github.com/forcedotcom/salesforcedx-vscode/pull/2893